algorithms/src/Kokkos_Sort.hpp

```diff
@@ -308,18 +315,32 @@ class BinSort {
                   values_range_begin - range_begin);
 
       parallel_for("Kokkos::Sort::CopyPermute",
-                   Kokkos::RangePolicy<execution_space>(0, len), functor);
+                   Kokkos::RangePolicy<ExecutionSpace>(exec, 0, len), functor);
     }
 
     {
       copy_functor<ValuesViewType, scratch_view_type> functor(
           values, range_begin, sorted_values);
 
       parallel_for("Kokkos::Sort::Copy",
-                   Kokkos::RangePolicy<execution_space>(0, len), functor);
+                   Kokkos::RangePolicy<ExecutionSpace>(exec, 0, len), functor);
     }
 
-    execution_space().fence("Kokkos::Sort: fence after sorting");
+    exec.fence("Kokkos::Sort: fence after sorting");
```

Comment:
Actually wondering if we should have the fence here at all

Reply:
For now, I just wanted to preserve the old behavior. I think moving the fence to the version not taking an execution space is fine, though, so we can declare the new variant to behave similarly to `deep_copy`. What do you think?

Reply:
I'd support moving the `fence()` call to the old overload, and letting this be completely asynchronous.

Reply:
Fixed in [`77129ce` (#4490)](https://github.com/kokkos/kokkos/pull/4490/commits/77129cedd8ce67aa4cb9b801fd55dec1f8285911).